contracts/token/ZNSDomainToken.sol

```diff
@@ -1,27 +1,27 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.18;
-
-import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { ERC721Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import { IZNSDomainToken } from "./IZNSDomainToken.sol";
 import { AccessControlled } from "../access/AccessControlled.sol";
 
 
 /**
  * @title A contract for tokenizing domains under ZNS
  */
-contract ZNSDomainToken is AccessControlled, ERC721, IZNSDomainToken {
+contract ZNSDomainToken is AccessControlled, UUPSUpgradeable, ERC721Upgradeable, IZNSDomainToken {
 
-    modifier onlyRegistrar() {
+    modifier onlyRegistrar {
         accessController.checkRegistrar(msg.sender);
         _;
     }
 
-
-    constructor(
+    function initialize(
+        address accessController,
         string memory tokenName,
-        string memory tokenSymbol,
-        address accessController
-    ) ERC721(tokenName, tokenSymbol) {
+        string memory tokenSymbol
+    ) public override initializer {
```

Comment:
why it's public here, but external in other contracts? let's bring them to the same visibility everywhere